video/out/vo_sixel.c

```diff
@@ -209,6 +209,8 @@ static SIXELSTATUS prepare_dynamic_palette(struct vo *vo)
 
         sixel_dither_set_diffusion_type(priv->dither, priv->opt_diffuse);
     } else {
+        if (priv->dither == NULL)
+            return SIXEL_FALSE;
```

Comment:
Who uses this return value? what happens if it fails?

Reply:
It looks like it is not being used, I could probably replace the functions to be void type and just return early instead. If it fails it should still not cause any issues because in `flip_page` we do check if the `dither` is NULL before calling sixel_encode.
And each time the `prepare_*_palette` function is called, dither and testdither will be re-initialized. It might keep failing in a loop though. I don't see any NULL pointer dereferencing occuring regardless in the current code flow assuming we have `draw_image` -> `prepare_palette` -> `flip_page` occuring in a linear fashion.

Reply:
OK. Let's keep the return value, but add a comment where it's _called_ that we can't use the return value, but otherwise the worst case is that the pointer will be checked before display and we will not render the image (or something more correct which will happen).

Reply:
Actually, let's check the return value where it's called, and print/log a warning using `MP_WARN` on such case (while still documenting the practical consequences of failure).

Reply:
Okay I've changed the function to void type because it gets called only in `draw_frame`. Also have added MP_ERR/MP_LOG for all places where `status` was being returned.
Only question I have is, do we have to check and honor for `--quiet` and -`--really-quiet` options? I'm assuming for now that MP_LOG and MP_ERR internally check for the options and output accordingly.

Reply:
I might have been unclear, sorry.

I meant to propagate the return value as long as we can, and once we can't do that anymore - `draw_imagw` - then print the warning.

> I'm assuming for now that MP_LOG and MP_ERR internally check for the options and output accordingly.

Yeah, that's OK, they will respect the quietness options.